subsys/net/l2/wifi/wifi_shell.c

```diff
@@ -1361,6 +1361,34 @@ static int cmd_wifi_ap_stations(const struct shell *sh, size_t argc,
 	return 0;
 }
 
+static int cmd_wifi_ap_sta_disconnect(const struct shell *sh, size_t argc,
+				      char *argv[])
+{
+	struct net_if *iface = net_if_get_first_wifi();
+	uint8_t mac[6];
+	int ret;
+
+	if (argc != 2) {
+		shell_fprintf(sh, SHELL_WARNING, "Invalid number of arguments\n");
+		shell_help(sh);
+		return -ENOEXEC;
+	}
```

Comment:
Isn't this check redundant with `SHELL_CMD_ARG()` usage, which handles the argument count validation on the shell side?

Reply:
Yeah, it can be removed, but having it also is okay in case we break that dependency. (True for all other commands as well).

Reply:
It's not a big deal, but IMO it's just a few bytes of code wasted (+ we now have two sources of truth for argc for the command).

Reply:
I can submit a new PR that removes these arg checks and add some notes.

Reply:
They were only few, so, did this as part of this PR itself.